sn_node/src/bin/safenode/main.rs

```diff
@@ -274,18 +290,73 @@ fn monitor_node_events(mut node_events_rx: NodeEventsReceiver, ctrl_tx: mpsc::Se
     });
 }
 
-async fn create_secret_key_file(path: impl AsRef<Path>) -> Result<tokio::fs::File, std::io::Error> {
-    let mut opt = tokio::fs::OpenOptions::new();
+fn init_logging(
+    log_output_dest: Option<LogOutputDestArg>,
```

Comment:
Just a small Q, why do we have to parse the `opt.log_output_dest` into `LogOutputDestArg` here and then pass it around?
Instead, could we directly send `opt.log_output_dest` to  `fn init_logging` to obtain the `LogOutputDest`?

Reply:
Hi Roland,

Thanks for the question. I'm sorry but I don't quite understand what you're asking here. Would you mind posting a code change suggestion?

Cheers,

Chris

Reply:
Sure Chris, I was suggesting to move the `parsing` logic to the `init_logging` function
This would allow us to bypass the intermediate `LogOutputDestArg` state.

```suggestion
fn init_logging(
    log_output_dest: Option<String>,
    peer_id: PeerId,
    format: Option<LogFormat>,
) -> Result<(String, Option<WorkerGuard>)> {
    let logging_targets = vec![
        ("safenode".to_string(), Level::INFO),
        ("sn_transfers".to_string(), Level::INFO),
        ("sn_networking".to_string(), Level::INFO),
        ("sn_node".to_string(), Level::INFO),
    ];

    let output_dest = if let Some(log_output_dest) = log_output_dest {
        match log_output_dest.as_str() {
            "stdout" => LogOutputDest::Stdout,
            "default" => {
                let path = dirs_next::data_dir()
                    .ok_or_else(|| eyre!("could not obtain data directory path".to_string()))?
                    .join("safe")
                    .join("node")
                    .join(peer_id.to_string())
                    .join("logs");
                LogOutputDest::Path(path)
            }
            // The path should be a directory, but we can't use something like `is_dir` to check
            // because the path doesn't need to exist. We can create it for the user.
            value => LogOutputDest::Path(PathBuf::from(value)),
        }
    } else {
        LogOutputDest::Stdout
    };

```

Reply:
Hi Roland,

Thanks. If you look at the definition for the `log_output_dest` argument, the parsing function is actually assigned as part of the definition. So what you would actually be suggesting would be to turn that argument into just a `String` rather than a 'special' type. I think we probably want to retain the use of the special type.